src/base/BaseActionsRouter.sol

```diff
@@ -5,10 +5,11 @@ import {IPoolManager} from "@uniswap/v4-core/src/interfaces/IPoolManager.sol";
 import {SafeCallback} from "./SafeCallback.sol";
 import {CalldataDecoder} from "../libraries/CalldataDecoder.sol";
 import {ActionConstants} from "../libraries/ActionConstants.sol";
+import {IMsgSender} from "../interfaces/IMsgSender.sol";
 
 /// @notice Abstract contract for performing a combination of actions on Uniswap v4.
 /// @dev Suggested uint256 action values are defined in Actions.sol, however any definition can be used
-abstract contract BaseActionsRouter is SafeCallback {
+abstract contract BaseActionsRouter is IMsgSender, SafeCallback {
```

Comment:
I think the msgSender function should be removed from BaseActionsRouter file then, if its being inherited from an interface?

Reply:
the virtual function itself is used by BaseActionsRouter, so I dont think we can remove it

```solidity
    function msgSender() public view virtual returns (address);

    /// @notice Calculates the address for a action
    function _mapRecipient(address recipient) internal view returns (address) {
        if (recipient == ActionConstants.MSG_SENDER) {
            return msgSender(); // <======
        } else if (recipient == ActionConstants.ADDRESS_THIS) {
            return address(this);
        } else {
            return recipient;
        }
    }
```

I think either BaseActionsRouter inherits `IMsgSender` or `IBaseActionsRouter is IMsgSender` -- the latter is weird because BAR's only external function IS `msgSender()` haha